test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/Listener/RequestBodyTests.cs

```diff
@@ -17,13 +17,40 @@ namespace Microsoft.AspNetCore.Server.HttpSys.Listener
     public class RequestBodyTests
     {
         [ConditionalFact]
+        public async Task RequestBody_SyncReadEnabledByDefault_ThrowsWhenDisabled()
+        {
+            string address;
+            using (var server = Utilities.CreateHttpServer(out address))
+            {
+                Task<string> responseTask = SendRequestAsync(address, "Hello World");
+
+                Assert.True(server.Options.AllowSynchronousIO);
+
+                var context = await server.AcceptAsync(Utilities.DefaultTimeout);
+                byte[] input = new byte[100];
+
+                Assert.True(context.AllowSynchronousIO);
+                var read = context.Request.Body.Read(input, 0, input.Length);
+                context.Response.ContentLength = read;
+                context.Response.Body.Write(input, 0, read);
+
+                context.AllowSynchronousIO = false;
+                Assert.Throws<InvalidOperationException>(() => context.Request.Body.Read(input, 0, input.Length));
+
+                string response = await responseTask;
+                Assert.Equal("Hello World", response);
+            }
+        }
+
+        [ConditionalFact]
         public async Task RequestBody_ReadSync_Success()
         {
             string address;
             using (var server = Utilities.CreateHttpServer(out address))
             {
                 Task<string> responseTask = SendRequestAsync(address, "Hello World");
 
+                server.Options.AllowSynchronousIO = true;
```

Comment:
Not at the moment, but they will be when we switch the default and since I had already fixed them there was no point in reverting it.